scripts/generate-gql-error-index-from-template.py

```diff
@@ -0,0 +1,184 @@
+
+            error_codes[error_code] = {
+                'description': description,
+                'page_role': page_role
+            }
+
+            if verbose and description:
+                print(f"Found description for {error_code}: {description[:50]}...")
+            elif verbose:
+                print(f"No description found for {error_code}")
+
+    # If we didn't find any descriptions, examine a sample file
+    if verbose and sample_file and not any(code['description'] for code in error_codes.values()):
+        print("\nNo descriptions were found in any files. Examining a sample file:")
+        examine_error_file(sample_file)
+
+    return error_codes
+
+def generate_from_template(template_file, errors_dir, output_file, include_descriptions=True, verbose=False):
+    """Generate the index file from template and individual error files."""
+    if verbose:
+        print("Extracting error codes and descriptions...")
+
+    # Get error codes and their info with improved extraction
+    error_codes = get_error_codes_from_files(errors_dir, verbose)
+
+    if verbose:
+        desc_count = sum(1 for code in error_codes.values() if code['description'])
+        print(f"Found {len(error_codes)} error code files, {desc_count} with descriptions")
+
+    # Read the template
+    with open(template_file, 'r') as f:
+        template_content = f.read()
+
+    # Define patterns to find placeholders
+    patterns = [
+        r'\{codes_starting_with:\'([^\']+)\'\}',
+        r'\{codes-starting with: \'([^\']+)\'\}'
+    ]
+
+    # Replace placeholders with generated content
+    for pattern in patterns:
+        for match in re.finditer(pattern, template_content):
+            prefix = match.group(1)
+            placeholder = match.group(0)
+
+            if verbose:
+                print(f"Processing prefix: {prefix}")
+
+            # Generate content for this prefix
+            content = []
+            matching_codes = [code for code in sorted(error_codes.keys()) if code.startswith(prefix)]
+
+            for error_code in matching_codes:
+                # Add page role if exists
+                if error_codes[error_code]['page_role']:
+                    content.append(f'[role=label--{error_codes[error_code]["page_role"]}]')
+
+                content.append(f'=== xref:errors/gql-errors/{error_code}.adoc[{error_code}]')
+                content.append('')
+
+                # Add description if available and requested
+                if include_descriptions and error_codes[error_code]['description']:
+                    content.append(f'Status description:: {error_codes[error_code]["description"]}')
+                    content.append('')
+
+            section_content = '\n'.join(content)
+
+            # Replace placeholder with generated content
+            template_content = template_content.replace(placeholder, section_content)
+
+    # Write the result to the output file
+    with open(output_file, 'w') as f:
```

Comment:
You can add the `// this is generated` comment to the file here with `f.write("// this is generated")`